Build: Let revapi compare API compatibility against apache-iceberg-1.0.0

ajantha-bhat opened the pull request:
Update to the latest release and enable CI running for just the `revapi.yml` file change

ajantha-bhat commented on .palantir/revapi.yml:
```diff
@@ -1,5 +1,5 @@
 versionOverrides:
-  org.apache.iceberg:iceberg-api:apache-iceberg-0.14.0: "0.14.0"
+  org.apache.iceberg:iceberg-api:apache-iceberg-1.0.0: "1.0.0"
```
not sure why didn't we have 0.14.1 before and does that tag version here need to be from the master? Currently, `apache-iceberg-1.0.0` is not from the master.

nastra replied:
looks like no CI was executed with this change, can you add `revapi.yml` file to https://github.com/apache/iceberg/blob/master/.github/workflows/api-binary-compatibility.yml#L29 so that CI runs?

ajantha-bhat replied:
Added. Thanks for the suggestion.

ajantha-bhat replied:
I think this failed now because the tag is not from the master? Looks like it even tried 0.14.0 but not 0.14.1

ajantha-bhat commented on .github/workflows/api-binary-compatibility.yml:
```diff
@@ -28,6 +28,7 @@ on:
   pull_request:
     paths:
       - 'api/**'
+      - '.palantir/revapi.yml'
```
enabling CI for just `revapi.yml `changes.

nastra commented:
@ajantha-bhat this should work:

```
-  org.apache.iceberg:iceberg-api:apache-iceberg-0.14.0: "0.14.0"
+  org.apache.iceberg:iceberg-api:1.0.0: "1.0.0"
```

ajantha-bhat commented:
> @ajantha-bhat this should work:
-  org.apache.iceberg:iceberg-api:apache-iceberg-0.14.0: "0.14.0"
+  org.apache.iceberg:iceberg-api:1.0.0: "1.0.0"

a. Could you please explain to me why do you think this can work?
b. It didn't work.

nastra commented:
Let's change this to the below diff then. That should work. I don't know why the same version definition doesn't work inside `revapi.yml`:
```
git diff .palantir/revapi.yml build.gradle                                                                                                                                                                                                       ──(Tue,Nov08)─┘
diff --git a/.palantir/revapi.yml b/.palantir/revapi.yml
index 2edd12342..615387108 100644
--- a/.palantir/revapi.yml
+++ b/.palantir/revapi.yml
@@ -1,5 +1,3 @@
-versionOverrides:
-  org.apache.iceberg:iceberg-api:apache-iceberg-0.14.0: "0.14.0"
 acceptedBreaks:
   apache-iceberg-0.14.0:
     org.apache.iceberg:iceberg-api:
diff --git a/build.gradle b/build.gradle
index 762f11573..be0a1fe49 100644
--- a/build.gradle
+++ b/build.gradle
@@ -227,6 +227,12 @@ project(':iceberg-bundled-guava') {
 project(':iceberg-api') {
   apply plugin: 'com.palantir.revapi'
 
+  revapi {
+    oldGroup = "org.apache.iceberg"
+    oldName = "iceberg-api"
+    oldVersion = "1.0.0"
+  }
+
```

ajantha-bhat commented:
> Let's change this to the below diff then. That should work. I don't know why the same version definition doesn't work inside revapi.yml:

Locally, I have changed and added new interface to test it out. Sadly, Interface change was not caught by revAPI. 
So, this might not be the right fix. 

@nastra: I have also added you as a collaborator to my repo. So, you can directly push to this branch, if you have any other ideas.  Thanks for thinking about this problem.

ajantha-bhat commented:
> Locally, I have changed and added new interface to test it out. Sadly, Interface change was not caught by revAPI. 
So, this might not be the right fix. 

the latest update, I added a default method in an interface, which won't be prompted by revAPI. 
So, everything works as before. 
Thanks @nastra !

rdblue commented:
Looks good to me! Thanks, @ajantha-bhat!
